include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -129,20 +129,41 @@ using projected_value_t = std::remove_cvref_t<std::invoke_result_t<Proj&, std::i
 namespace __ranges
 {
 
-struct __first_size_calc
+template <std::size_t _RngIndex>
+struct __eval_rng_size
```

Comment:
What do you think about it?

```c++
template <typename _Range>
constexpr auto
__size(const _Range& __range)
{
#if CPP20_RANGES_PRESENT
        return std::ranges::size(__range);
#else
        return __range.size();
#endif
}

template <std::size_t _RangeIndex>
struct __nth_range_size
{
    template<typename... _Ranges>
    auto operator()(const _Ranges&... __ranges)
    {
        static_assert(RangeIndex < sizeof...(_Ranges));
        return __size(std::get<_RangeIndex>(std::tie(__ranges...)));
    }
};

using __first_size_calc = __nth_range_size<0>;
using __second_size_calc = __nth_range_size<1>;
```

- Separate `__size` to help with #2445. 
- Simpler implementation without recursion (in this code, at least).
- `__eval_rng_size` -> `__nth_range_size`.